src/openrct2/world/TileElement.cpp

```diff
@@ -168,7 +168,29 @@ void TileElement::ClearAs(uint8_t newType)
 
 void TileElementBase::Remove()
 {
-    tile_element_remove(static_cast<TileElement*>(this));
+    auto tileElement = this;
```

Comment:
This is the logic that use to live in `tile_element_remove`. It seems to make more sense to have it here. Let me know if I'm off base here.